File: packages/alteration/src/main/java/fdit/alteration/core/incident/WayPoint.java
package fdit.alteration.core.incident;
import com.fasterxml.jackson.annotation.JsonProperty;
public class WayPoint {

    @JsonProperty("vertex")
    private Vertex vertex;

    @JsonProperty("altitude")
    private Altitude altitude;

    @JsonProperty("time")
    private long time;

    public Vertex getVertex() {
        return vertex;
    }

    public void setVertex(final Vertex vertex) {
        this.vertex = vertex;
    }

    public Altitude getAltitude() {
        return altitude;
    }

    public Integer getAltitudeValue() {
        return altitude.getContent();
    }

    public Boolean isAltitudeOffset() {
        return altitude.getOffset();
    }

    public Boolean isTimedAltitude() {
        return altitude.getTime();
    }

    public Boolean isTimedPosition() {
        return vertex.getTime() != null;
    }

    public void setAltitude(final Altitude altitude) {
        this.altitude = altitude;
    }

    public long getTime() {
        return time;
    }

    public void setTime(final long time) {
        this.time = time;
    }
}

File: packages/fdit-ts/src/Incident/WayPoint.ts
import { Vertex } from './Vertex'
import { Altitude } from './Altitude'

export class WayPoint {
  private vertex?: Vertex
  private altitude?: Altitude
  private time: number = 0

  public getVertex(): Vertex | undefined {
    return this.vertex
  }

  public setVertex(vertex: Vertex): void {
    this.vertex = vertex
  }

  public getAltitude(): Altitude | undefined {
    return this.altitude
  }

  public getAltitudeValue(): number {
    return this.altitude?.getContent() ?? 0
  }

  public isAltitudeOffset(): boolean {
    return this.altitude?.getOffset() ?? false
  }

  public isTimedAltitude(): boolean {
    return !!this.altitude?.getTime()
  }

  public isTimedPosition(): boolean {
    return !!this.vertex?.getTime()
  }

  public setAltitude(altitude: Altitude): void {
    this.altitude = altitude
  }

  public getTime(): number {
    return this.time
  }

  public setTime(time: number): void {
    this.time = time
  }
}
